### PR TITLE
Wu signandlock

### PR DESCRIPTION
**Pull request by nakul2234**
When attendings make a workup, it is automatically signed by them. Once an attending signs a workup, only other attendings can see the update button for workups; coordinators, clinicals, and pre-clinicals cannot.

### CODE REVIEW

**Comment by justinrporter** [comment]
Awesome! I will look at this tomorrow after my pharm exam and do a code review and we'll go from there. Nice work being so productive!

File: pttrack/templates/pttrack/workup_detail.html [hunk 1]
```diff
@@ -22,7 +22,9 @@
     <div class="row text-center">
         <h4>Actions</h4>
         <div class="btn-group">
+            {% if workup.signed and request.session.clintype_pk == "Attending" or not workup.signed %}
```
Comment (justinrporter) [inline pttrack/templates/pttrack/workup_detail.html]: I am wondering if we could come up with a way of doing this that doesn't require a string comparison? Could we do like `request.session.clintype.signs_charts` or something? I don't really remember the interface...

**Comment by justinrporter** [comment]
So, it looks like your solution to deal with updates-after-attestation works to prevent people from clicking a button to get the the update URL. *But*, if I just punch in the correct URL, I can still get to the update page! I think that it would be good to add a check to the WorkupUpdate class view to prevent the site from even serving the update page *and* from accepting a POST with new data (for example, if I already have a copy of the update page I will still be able to submit my changes).

There should be a method in the FormUpdate class-based-view that you can override to do that. I think it's maybe `request`?

Thoughts?

File: pttrack/templates/pttrack/workup_detail.html (hunk 1 above)
Comment (nakul2234) [inline pttrack/templates/pttrack/workup_detail.html]: I get what you are saying, but I am not sure if that will work. I'll look into a work around

**Comment by nakul2234** [comment]
@justinrporter i decided to create a separate error page for when this happens. I decided to make a template because if the software gets more advanced, we could have options allowing the user to somehow send the changes they were suggesting to the attending who signed so they can change it later if there was a mistake or important update they wanted the attending to know.

**Comment by justinrporter** [comment]
> i decided to create a separate error page for when this happens

This is a great idea not just for the future possibility of a "suggest change" feature, but also because it informs the user what went wrong, which is always a good idea.

> Also, I made another commit to add the functionality of having an email field when the provider is created, and then having their first name, last name, and email being copied into the User model.

I am not sure that this is the right way to do this. You should always be able to access the User model from the Provider with `provider.associated_user` (and the Provider from the User with `user.provider`). What is the purpose of duplicating this information? You should be able to add an extra email field this to the `ProviderForm` or something rather than duplicating the information in the model... We can chat about this at clinic tomorrow.

**Comment by nakul2234** [comment]
@justinrporter I made the extra email field and got rid of the temporary provider_email variable in the provider class. It is up to date with the main branch and should work.

File: pttrack/views.py [hunk 2]
```diff
@@ -427,9 +444,15 @@ def byName_key(patient):
                    'title': pagetitle})
 
 
+def error_workup(request, pk):
+
+    wu = get_object_or_404(mymodels.Workup, pk=pk)
+    return render(request,
+                  'pttrack/workup_error.html',
+                  {'workup': wu})
```
Comment (justinrporter) [inline pttrack/views.py]: You could use a detail view for this. You just provide the template, and the detail view takes care of doing all of this for you.

**Comment by justinrporter** [comment]
Awesome work! Is it tested?

**Comment by nakul2234** [comment]
Yeah tested it a few times and it is working. Ill merge it, but I will leave the branch for now because we may want to change a few things relating to Workup premissinso based on how smoothly this goes.

**Pull request merged**